store/src/java/com/zimbra/soap/SoapEngine.java

```diff
@@ -327,6 +332,22 @@ private Element dispatch(String path, Element envelope, Map<String, Object> cont
         Element ectxt = soapProto.getHeader(envelope, HeaderConstants.CONTEXT);
         try {
             zsc = new ZimbraSoapContext(ectxt, doc.getQName(), handler, context, soapProto);
+            if (doc.getName().equals("SendMsgRequest")) {
+                Element msgEle = doc.getElement(MailConstants.E_MSG);
+                String identityId = msgEle.getAttribute(MailConstants.A_IDENTITY_ID, null);
+                if (identityId != null) {
+                    Identity identity = Provisioning.getInstance().get(DocumentHandler.getOperationContext(zsc, context)
+                            .getAuthenticatedUser(), Key.IdentityBy.id, identityId);
+                    if (identity != null &&
+                            !identity.getAttr(Provisioning.A_zimbraPrefIdentityName).equals(ProvisioningConstants.DEFAULT_IDENTITY_NAME) &&
+                            !StringUtil.isNullOrEmpty(identity.getAttr(Provisioning.A_zimbraPrefFromAddressType)) &&
+                            (identity.getAttr(Provisioning.A_zimbraPrefFromAddressType).equals(RightConsts.RT_sendAs) ||
+                                    identity.getAttr(Provisioning.A_zimbraPrefFromAddressType).equals(RightConsts.RT_sendOnBehalfOf))) {
+                        Account targetAccount = Provisioning.getInstance().get(AccountBy.name, identity.getAttr(Provisioning.A_zimbraPrefFromAddress));
+                        zsc.setmRequestedAccountId(targetAccount.getId());
```

Comment:
Maybe not a big deal but if `requestedAccountId` is already set to something else, this would override that.

Reply:
Only if the above condition satisfied, then only overriding the value of `requestedAccountId`